src/main/webapp/scripts/sessioncache.js

```diff
@@ -0,0 +1,117 @@
+/** 
+ * SessionCache bridges the gap between the client and server.
+ * Allows client to indirectly stay in contact with the server,
+ * encapsulating many of the contact points away from the client.
+ * 
+ * SessionCache looks for information about the current session and
+ * updates the server with information about each client in return (like
+ * date last contacted). Also includes some additional functionality, 
+ * like contacting the server to update the controller of the session.
+ */
+class SessionCache {
+  /**
+   * Initalizes a SessionCache object.
+   * @param {Object} urlParams Represents the URLSearchParams of the
+   *    session the client is in, holds information such as the
+   *    session ID and the screen name of the current user.
+   */
+  constructor(urlParams) {
+    /** 
+     * Poll responsible for contacting the server for information about
+     * the current session.
+     * @private {Object} 
+     */
+    this.sessionInformationPoll_ = null;
+
+    /** 
+     * Poll responsible for updating the server of information about Date
+     * last polled. 
+     * @private {Object} 
+     */
+    this.trackForInactivityPoll_ = null;
+
+    /**
+     * Holds the different keys being tracked by the SessionCache and 
+     * their respective values.
+     * @private {Object}
+     */
+    this.cache_ = { session: null };
```

Comment:
the value of session will eventually be an object of Session

Reply:
Is this cache meant to be a general cache or a cache specifically for a session? If the former, then some of these variables are too specific to the project. If the latter, then you dont need this to be a whole object, right? It can be a single value.